lib/airbrake-ruby/filters/git_last_checkout_filter.rb

```diff
@@ -0,0 +1,85 @@
+require 'strscan'
+
+module Airbrake
+  module Filters
+    # Attaches git checkout info to `context`. The info includes:
+    #   * username
+    #   * email
+    #   * revision
+    #   * time
+    #
+    # This information is used to track deploys automatically.
+    #
+    # @api private
+    # @since v2.12.0
+    class GitLastCheckoutFilter
+      # @return [Integer]
+      attr_reader :weight
+
+      # @param [Logger] logger
+      # @param [String] root_directory
+      def initialize(logger, root_directory)
+        @git_path = File.join(root_directory, '.git')
+        @logger = logger
+        @weight = 116
+        @last_checkout = nil
+      end
+
+      # @macro call_filter
+      def call(notice)
+        return if notice[:context].key?(:lastCheckout)
+
+        if @last_checkout
+          notice[:context][:lastCheckout] = @last_checkout
+          return
+        end
+
+        return unless File.exist?(@git_path)
+        return unless (checkout = last_checkout)
+        notice[:context][:lastCheckout] = checkout
+      end
+
+      private
+
+      def last_checkout
+        return unless (line = last_checkout_line)
+
+        parts = line.chomp.split("\t").first.split(' ')
+        if parts.size < 5
+          @logger.error(
+            "#{LOG_LABEL} Airbrake::#{self.class.name}: can't parse line: #{line}"
+          )
+          return
+        end
+
+        author = parts[2..4]
```

Comment:
I am not sure you can use `4` here, because this number will be different if user name contains space or not, e.g. "vmihailenco" or "Vladimir Mihailenco".

Reply:
Yeah, looks like you're right. It'll be broken.